### PR TITLE
Misc fixes for ink! 3.0-rc1

### PR DESCRIPTION
**Pull request by Robbepop**
Closes https://github.com/paritytech/ink/issues/519.

### CODE REVIEW

File: crates/allocator/Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ edition = "2018"
 license = "Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/paritytech/ink"
-documentation = "https://substrate.dev/substrate-contracts-workshop/#/"
+documentation = "https://docs.rs/ink_allocator/"
```
Comment (cmichi) [inline crates/allocator/Cargo.toml:10]: Just noted that https://docs.rs/ink_allocator/ is completely empty. Maybe at least a module documentation comment would be good?

File: crates/storage/src/alloc/boxed/storage.rs [hunk 2]
```diff
@@ -29,21 +29,27 @@ use ink_primitives::Key;
...
+                    scale_info::Path::from_segments(vec!["ink_storage", "alloc", "Box"])
+                        .expect("encountered invalid Rust path"),
+                )
+                // Unfortunately we cannot encode the type parameters of the box since they
+                // have to be `T: scale::Codec`. However, them not requiring to be encodable
+                // is the purpose of the storage `Box<T>`.
+                // Until we found a solution to this problem we cannot uncomment the below
+                // line of code:
+                //
+                // .type_params(vec![scale_info::MetaType::new::<T>()])
```
Comment (cmichi) [inline crates/storage/src/alloc/boxed/storage.rs:48]: What's the trade-off of not being able to un-comment the line?

Comment (Robbepop) [reply crates/storage/src/alloc/boxed/storage.rs:66]: It does not make sense for the `T` in `ink_storage::alloc::Box<T>` to implement `scale_info::TypeInfo` so having this requirement on the bounds would not make sense, too. The `T: scale_info::TypeInfo` bound means that `T` is somehow `scale` encodable. However, `ink_storage::alloc::Box<T>` actually exists exactly for the purpose of allowing non-`scale` encodable instances within others.

**Review by cmichi — APPROVED** [review]
LGTMichi :-)

**Pull request merged**